java/api/src/main/java/org/ray/api/options/ActorCreationOptions.java

```diff
@@ -9,7 +9,7 @@
 public class ActorCreationOptions extends BaseTaskOptions {
 
   public static final int NO_RECONSTRUCTION = 0;
-  public static final int INFINITE_RECONSTRUCTIONS = (int) Math.pow(2, 30);
+  public static final int INFINITE_RECONSTRUCTION = (int) Math.pow(2, 30);
```

Comment:
This was inconsistent between Python and Java.

Reply:
@deanwampler We just use a very large number to imitate "infinite". In this way we could simplify the implementation. I agree this is not perfect. But users don't have to care the actual value of this constant. So I think this is fine. 
Regarding the name, I think it's clearer when users specify `setMaxReconstructions(INFINITE_RECONSTRUCTION)`. `setMaxReconstructions(MAX_RECONSTRUCTION)` is a bit unclear.

Reply:
Also, based on our experiment. In real applications, our users never want to use a finite number. Their choice is always 0 or infinite. So I'm thinking of adding a `enableAutoReconstruction` API, which just sets max_reconstructions = infinite. 
And we could still keep the `setMaxReconstructions` API for tests and potential power users. 
Do you think it will be clearer this way? @robertnishihara @deanwampler

Reply:
I don't like `enableAutoReconstruction` very much because it introduces more configuration flags that interact with each other. I think `max_reconstructions=ray.ray_constants.INFINITE_RECONSTRUCTION` is fine for now.

Reply:
Okay, I don't have strong preference either.  Let's keep the current way for now.